### PR TITLE
fix(client): send valid headers dataproxy

### PR DESCRIPTION
**Pull request by millsp**
Test here (failing before this merge) https://github.com/prisma/e2e-tests/pull/2379

### CODE REVIEW

File: packages/engine-core/src/data-proxy/utils/request.ts [hunk 1]
```diff
@@ -85,6 +86,7 @@ function nodeFetch(url: string, options: RequestOptions = {}): Promise<RequestRe
   return new Promise((resolve, reject) => {
     const https: typeof _https = eval(`require('https')`)
 
+    console.log(httpsOptions)
```
Comment (millsp) [inline packages/engine-core/src/data-proxy/utils/request.ts:89]:
```suggestion
```

File: packages/engine-core/src/data-proxy/utils/request.ts [hunk 2]
```diff
@@ -38,7 +38,8 @@ export async function request(url: string, options: RequestOptions = {}): Promis
  */
 function buildHeaders(options: RequestOptions): RequestOptions['headers'] {
   return {
-    ...options.headers,
+    // this ensures headers will always be valid
+    ...JSON.parse(JSON.stringify(options.headers)),
```
Comment (Jolg42) [inline packages/engine-core/src/data-proxy/utils/request.ts:42]: Which value was passed that prompted this? 👀

Comment (millsp) [reply packages/engine-core/src/data-proxy/utils/request.ts:42]: `undefined`

Comment (millsp) [reply packages/engine-core/src/data-proxy/utils/request.ts:42]: Discarded correctly everywhere besides the custom nodeFetch

Comment (Jolg42) [reply packages/engine-core/src/data-proxy/utils/request.ts:42]: I see https://github.com/prisma/e2e-tests/pull/2379#issuecomment-1026656733

**Review by Jolg42 — APPROVED** [review]

**Pull request merged**